src/util/command-template.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022-present Mirage Aegis
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { SlashCommandBuilder, EmbedBuilder, ChatInputCommandInteraction } from "discord.js";
+import { ErrorHandler } from "./error-handler.js";
+
+export type Command = {
+    /**
+     * Metadata for the slash command.
+     * Must include name, description, and DM permissions.
+     * May also include command options
+     */
+    readonly data: SlashCommandBuilder;
+
+    /**
+     * The callback function to execute whenever a slash command is used
+     * 
+     * @param ctx the command context
+     */
+    execute(ctx: ChatInputCommandInteraction): Promise<void>;
+
+
+    readonly error: ErrorHandler;
+
+    /**
+     * The embed to display when help regarding the command is requested.
+     * Must include title, description, and appropriate fields (such as format)
+     */
+    readonly help: EmbedBuilder;
+};
```

Comment:
This is the structure that each command **MUST** follow. The command must have metadata, an execution, an error handler, and an embed for the `/help` command